### PR TITLE
Fix shortcut related bugs

### PR DESCRIPTION
**Pull request by StanleyW00**
Fix some bugs in shortcut feature.

### CODE REVIEW

File: src/main/java/essenmakanan/exception/EssenEditShortcutException.java [hunk 1]
```diff
@@ -17,6 +17,9 @@ public void handleException() {
         case "same name":
             System.out.println("You can't edit shortcut's name with the same name.");
             break;
+        case "same quantity":
+            System.out.println("You can't edit shortcut's quantity with the same quantity.");
+            break;
         case "usage":
             System.out.println("You can't edit a shortcut's property more than once in a single line.");
             break;
```
Comment (kaijie0102) [inline src/main/java/essenmakanan/exception/EssenEditShortcutException.java:25]: great way to handle exceptions! i like it

File: src/main/java/essenmakanan/parser/RecipeParser.java [hunk 2]
```diff
@@ -218,16 +218,23 @@ public static RecipeIngredientList parseDataRecipeIngredients(String ingredients
...
-            IngredientUnit ingredientUnit = IngredientUnit.valueOf(parsedIngredient[2]);
+            String ingredientName = parsedIngredient[0].trim();
+
+            for (Ingredient ingredient : ingredientList) {
+                if (ingredient.getName().equals(ingredientName)) {
+                    throw new EssenStorageFormatException();
+                }
+            }
+
+            Double ingredientQuantity = Double.parseDouble(parsedIngredient[1].trim());
```
Comment (kaijie0102) [inline src/main/java/essenmakanan/parser/RecipeParser.java:236]: would it be safer to check if ingredientQuantity is parsable? Perhaps in a try catch `NumberFormatException`

File: src/main/java/essenmakanan/parser/ShortcutParser.java [hunk 3]
```diff
@@ -69,25 +73,37 @@ private static void editShortcutName(Shortcut shortcut, IngredientList ingredien
...
-            throws EssenEditShortcutException {
+            throws EssenEditShortcutException, NumberFormatException {
         double newQuantity = Double.parseDouble(editDetail.substring(2).strip());
 
         if (hasEditQuantity) {
             throw new EssenEditShortcutException("usage");
         }
 
+        if (!checkForValidQuantity(newQuantity)) {
+            throw new NumberFormatException();
```
Comment (kaijie0102) [inline src/main/java/essenmakanan/parser/ShortcutParser.java:84]: should we be keeping the exceptions to our Essen exceptions only?

**Review by kaijie0102 — CHANGES_REQUESTED** [review]
overall, LGTM, some minor changes only

File: src/main/java/essenmakanan/parser/ShortcutParser.java (hunk 3 above)
Comment (StanleyW00) [reply src/main/java/essenmakanan/parser/ShortcutParser.java:84]: this will trigger essen exception after throwing it

**Review by kaijie0102 — APPROVED** [review]
LGTM! Thanks for changes and clarifications!

**Pull request merged**